File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <queue>

using namespace std;

bool solution(string s)
{
    bool answer = true;

    queue<int> q;
    for(char c: s){
        if(c == '(') q.push(1);
        else if(c == ')') {
            if (q.empty()){
                answer = false;
                break;
            }
            else q.pop();
        }
    }
    
    if(!q.empty()) answer = false;

    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for c in s:
        if len(stack) == 0 and c == ')':
            return False
        elif len(stack) == 0 or c == stack[-1]:
            stack.append(c)
        else:
            stack.pop()
    
    answer = True if len(stack) == 0 else False

    return answer